### PR TITLE
Adding Guarda Wallet to Bitcoin.org

### PR DESCRIPTION
**Pull request by guardawallet**

### CODE REVIEW

**Comment by crwatkins** [comment]
@guardawallet Thanks for the submission!

The first step is get a clean Travis CI build.   Take a look at the log files for what is failing.  The error messages are hopefully self-explanatory about what is wrong with the syntax, but if not please ask about them here.

File: _wallets/guarda.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    name: desktop
+    default: &DEFAULT
+      text: "walletguarda"
+      link: "https://guarda.co/"
+      source: "https://github.com/guardaco/"
+      screenshot: "guarda.png"
+      features: "bech32 hardware_wallet legacy_addresses multisig segwit"
+      check:
+        control: "checkgoodcontrolfull"
+        validation: "checkgoodvalidationspvservers"
```
Comment (wbnns) [inline _wallets/guarda.md]: This isn't an accepted value for `validation`.

File: _wallets/guarda.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    os:
+      - name: android
+        text: "walletguarda"
+        link: "https://play.google.com/store/apps/details?id=com.crypto.multiwallet"
+        source: "https://github.com/guardaco/"
+        screenshot: "guarda.png"
+        features: "bech32 legacy_addresses segwit"
+        check:
+          control: "checkgoodcontrolfull"
+          validation: "checkgoodvalidationspvservers"
```
Comment (wbnns) [inline _wallets/guarda.md]: This isn't an accepted value for `validation`.

File: _wallets/guarda.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          privacynetwork: "checkpassprivacynetworksupporttorproxy"
+      - name: ios
+        text: "walletguarda"
+        link: "https://apps.apple.com/app/apple-store/id1442083982"
+        source: "https://github.com/guardaco/"
+        screenshot: "guarda.png"
+        features: "bech32 legacy_addresses segwit"
+        check:
+          control: "checkgoodcontrolfull"
+          validation: "checkgoodvalidationspvservers"
```
Comment (wbnns) [inline _wallets/guarda.md]: This isn't an accepted value for `validation`.

File: _wallets/guarda.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    os:
+      - name: web
+        text: "walletguarda"
+        link: "https://guarda.co/"
+        source: "https://github.com/guardaco/"
+        screenshot: "guarda.png"
+        features: "bech32 legacy_addresses hardware_wallet segwit multisig"
+        check:
+          control: "checkgoodcontrolfull"
+          validation: "checkgoodvalidationcentralized"
```
Comment (wbnns) [inline _wallets/guarda.md]: This isn't an accepted value for `validation`.

File: _wallets/guarda.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          transparency: "checkpasstransparencyremote"
```
Comment (wbnns) [inline _wallets/guarda.md]: This isn't an accepted value for `transparency`.

File: _wallets/guarda.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+          environment: "checkpassenvironmentdesktop"
```
Comment (wbnns) [inline _wallets/guarda.md]: This isn't an accepted value for `environment`.

**Comment by guardawallet** [comment]
Thank you for your comments. Now we will get a clean Travis CI build.

**Comment by crwatkins** [comment]
@guardawallet The source attributes in this submission point to a GitHub organization (which currently contains 79 repositories) and does not point to specific source code.  Could you update these attributes at a minimum to point to specific repositories?

Also, I have seen references in the these repositories to single currency wallets.  Am I correct that you are requesting the listing of a multi-currency wallet and not a single-currency BTC wallet?  Do you support one?  I suspect many Bitcoin users would prefer to avoid the attack surface of supporting 45 other currencies.

**Comment by IGORADEK** [comment]
New Order World Ilumination Diamond
MULTIUNIVERS

**Comment by guardawallet** [comment]
@crwatkins, thank you for your feedback. Looking back at our commit, we agree that we can mislead your user base.

We've got several products, a multicurrency web wallet, a multicurrency desktop wallet, a multicurrency android wallet, a multicurrency iOS wallet, multicurrency chrome extension wallet, and single-currency android wallets. We don't open-source our web and desktop wallets for the security reason for making harder fishing attempts, although the Web code is available through a web browser console. We are contributing to the open-source community by publishing useful insights and creating a public library of crypto icons.

Currently, our single currency wallets are not getting marketing publicity for the reason of a strike for publishing several single-currency wallets from our account on Google Play.

On Bitcoin.org we would like to list our multicurrency wallet because we provide 24/7 user support, constantly updating it and provide several Bitcoin features for BTC user base. For example, we support a multi-signature feature, fee editing, support for Omni tech and a Bitcoin testnet support.

We already fixed our pull request, so please contact us if you have any questions or comments.

**Comment by crwatkins** [comment]
@guardawallet Thanks for the explanation.    Bitcoin.org does not list wallets that do not have publicly reviewable source code.   The community has been fairly adamant about that requirement.  If you are still interested in listing any of your wallets which are open source, please review the [listing requirements](https://github.com/bitcoin-dot-org/bitcoin.org/blob/master/docs/managing-wallets.md) very carefully to make sure you can meet each and every one of them.   If you have any questions about any of them, please ask about them here.  Be sure to review the address reuse requirements as your multi-currency wallet appears to reuse  BTC addresses.

**Comment by guardawallet** [comment]
@crwatkins, thank you for your answer. We have gone through all the requirements again and found our single currency BTC wallet most suitable for your website. So we have updated the information in the pull request. The wallet we are including there is a non-custodial, android light wallet.

Just a quick question, will the description of a wallet also mention a link to our website?

Thank you again for your feedback.

**Comment by crwatkins** [comment]
@guardawallet 

1.  The `compat` line in your submission needs to be updated to reflect only Android

2. I've send funds to the wallet yet the receive address in the wallet does not change.   As I mentioned above, we have requirements on address re-use.  Do you update receive addresses?  What derivation path are you using?  It doesn't seem to match any of the common ones.

> Just a quick question, will the description of a wallet also mention a link to our website?

The listing will only include the URLs in your submission.

**Comment by crwatkins** [comment]
@guardawallet I'm having a lot of problems using this wallet

1.  I sent a transaction to the wallet 9 hours ago and then tried to make a payment with the wallet.  I used the MAX button to choose the amount.   I got a red warning (or error?)  that said "Amount of sending is very small".  I had no idea if that was a warning, or how to fix it, or how small "very small" is.  I now assume it may have something to do with a the unconfirmed transaction.

2.  There have now been 27 confirmations yet the wallet displays -119 confirmations.  I have no idea what -119 conformations mean.  

3.  I occasionally get messages saying "Transaction history is temporarily unavailable" and then it goes away.   This is an SPV wallet, right?  I'm not sure what that message means.

The bottom line is I sent funds to the wallet (other wallets on the same device are working fine) and I no longer seem to have access to those funds which is very disturbing.

**Comment by IGORADEK** [comment]
*ANUNNAKIGORADEKINGODIAMONDSANUNNAKI.G.O.RA.OM.ORION*
SPAM WALLET

sri, 4. ožu 2020. u 00:14 Craig Watkins <notifications@github.com> napisao
je:

**Comment by guardawallet** [comment]
> @guardawallet I'm having a lot of problems using this wallet
> 
>     1. I sent a transaction to the wallet 9 hours ago and then tried to make a payment with the wallet.  I used the MAX button to choose the amount.   I got a red warning (or error?)  that said "Amount of sending is very small".  I had no idea if that was a warning, or how to fix it, or how small "very small" is.  I now assume it may have something to do with the unconfirmed transaction.

We use this library, https://github.com/bitcoinj/bitcoinj, probably this is a reason for an error, also there is a possibility that you didn't choose to include a transaction fee. For example, we've successfully sent 0.000041 with a fee of 0.00001.

>     2. There have now been 27 confirmations yet the wallet displays -119 confirmations.  I have no idea what -119 conformations mean.

Dev team said they are fixing it already.

>     3. I occasionally get messages saying "Transaction history is temporarily unavailable" and then it goes away.   This is an SPV wallet, right?  I'm not sure what that message means.

We use the history from block explorer, sometimes it may be not available. You can change your connection from Wi-Fi to mobile network.

> The bottom line is I sent funds to the wallet (other wallets on the same device are working fine) and I no longer seem to have access to those funds which is very disturbing.

This is an SPV wallet, keys are not being sent anywhere, they are stored only on the user device. The transaction is signed locally and being sent to a full node.

Thank you for your test and comments, we are gonna apply fixes.

**Comment by crwatkins** [comment]
@guardawallet Even after the incoming funds confirmed, I was still unable to send the transaction.  It does not seem like this wallet is ready for use yet.    Your excellent support folks claimed there may be problems because  this wallet was "less supported".  They recommended I use the web version to recover my funds, which I did successfully.

The wallet re-uses addresses which does not meet our  listing requirements.

I would recommend resubmitting this wallet after the wallet meets all of the requirements and it has had some significant usage (another one of our requirements) in about nine months to a year.
